Library/brew.sh

```diff
@@ -230,6 +230,15 @@ then
     & disown
 fi
 
+if [[ -n "$HOMEBREW_AUTO_UPDATE" ]] &&
+   [[ "$HOMEBREW_COMMAND" = "install" || "$HOMEBREW_COMMAND" = "upgrade" ]]
+then
+  # Hide shellcheck complaint:
+  # shellcheck source=/dev/null
+  source "$HOMEBREW_LIBRARY/Homebrew/cmd/update.sh"
+  HOMEBREW_PRECOMMAND="homebrew-update --preinstall"
+fi
```

Comment:
I think it would be a bit cleaner to package this as a function like so:

```shell
update-before-install() {
  [[ -n "$HOMEBREW_AUTO_UPDATE" ]] || return

  if [[ "$HOMEBREW_COMMAND" = install || "$HOMEBREW_COMMAND" = upgrade ]]
  then
    # Hide shellcheck complaint:
    # shellcheck source=/dev/null
    source "$HOMEBREW_LIBRARY/Homebrew/cmd/update.sh"
    homebrew-update --preinstall
  fi
}
```

Then use that further down below, e.g.:

```shell
{ update-before-install; "homebrew-$HOMEBREW_COMMAND" "$@"; exit $?; }
```

To me that not only looks cleaner, but also clearer. It also avoids an unquoted variable expansion.